packages/core/src/tools/sanitize.spec.ts

```diff
@@ -96,6 +97,25 @@ describe('sanitize', () => {
         expect(sanitize(map)).toBe('[Map]')
       }
     })
+
+    it('should survive an improperly implemented toStringTag', () => {
+      function Func() {
+        /* empty */
+      }
+
+      Object.defineProperty(Func.prototype, Symbol.toStringTag, {
+        get: () => {
+          throw Error('Cannot serialize')
+        },
+        enumerable: false,
+        configurable: true,
+      })
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      const func = new (Func as any)()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏What about using a `class` in order to avoid the `any` cast and the eslint exception?